packages/tools/examples/segmentBidirectionalTool/findLargestBidirectional.ts

```diff
@@ -0,0 +1,218 @@
+    if (Math.abs(distance - length1 - length2) < 5 * EPSILON) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function createBidirectionalForSlice(sliceContour, currentMax, isInSegment) {
+  const { points } = sliceContour.polyData;
+  const { maxMinor: currentMaxMinor, maxMajor: currentMaxMajor } =
+    currentMax || { maxMajor: 0, maxMinor: 0 };
+  let maxMajor = currentMaxMajor * currentMaxMajor;
+  let maxMinor = currentMaxMinor * currentMaxMinor;
+  let maxMajorPoints;
+  for (let i = 0; i < points.length; i++) {
+    for (let j = i + 1; j < points.length; j++) {
+      const pointI = points[i];
+      const pointJ = points[j];
+      const distance2 = distanceSq(pointI, pointJ);
+      if (distance2 < maxMajor) {
+        continue;
+      }
+      if (distance2 === maxMajor && maxMajorPoints) {
+        // Consider adding to the set of points rather than continuing here
+        // so that all minor axis can be tested
+        continue;
+      }
+      if (!isInSegment(pointI, pointJ)) {
+        // Center between the two points has to be in the segment, otherwise
+        // this is out of bounds.
+        continue;
+      }
+      const distance = Math.sqrt(distance2);
+      if (isCrossing(points, pointI, pointJ, distance)) {
+        // If the line intersects the segment boundary, then skip it
+        continue;
+      }
+      maxMajor = distance2;
+      maxMajorPoints = [i, j];
+      maxMinor = 0;
+    }
+  }
+  if (!maxMajorPoints) {
+    return;
+  }
+
+  maxMajor = Math.sqrt(maxMajor);
+  const handle0 = points[maxMajorPoints[0]];
+  const handle1 = points[maxMajorPoints[1]];
+  const majorDelta = vec3.sub(vec3.create(), handle0, handle1);
+  majorDelta.forEach((it, index) => (majorDelta[index] = it / maxMajor));
+
+  let maxMinorPoints;
+
+  for (let i = 0; i < points.length; i++) {
+    for (let j = i + 1; j < points.length; j++) {
+      const pointI = points[i];
+      const pointJ = points[j];
+      const distance2 = distanceSq(pointI, pointJ);
+      if (distance2 <= maxMinor) {
+        continue;
+      }
+      const distance = Math.sqrt(distance2);
+      const delta = vec3.sub(vec3.create(), pointI, pointJ);
+
+      const dot = Math.abs(vec3.dot(delta, majorDelta)) / distance;
+      if (dot > EPSILON) {
+        continue;
+      }
+      if (!isInSegment(pointI, pointJ)) {
+        // Center between the two points has to be in the segment, otherwise
+        // this is out of bounds.
+        continue;
+      }
+      if (isCrossing(points, pointI, pointJ, distance)) {
+        continue;
+      }
+      maxMinor = distance2;
+      maxMinorPoints = [i, j];
+    }
+  }
+
+  if (!maxMinorPoints) {
+    // Didn't find a larger minor distance
+    return;
+  }
+  maxMinor = Math.sqrt(maxMinor);
+  const handle2 = points[maxMinorPoints[0]];
+  const handle3 = points[maxMinorPoints[1]];
+
+  const bidirectional = {
+    handle0,
+    handle1,
+    handle2,
+    handle3,
```

Comment:
since there is no order, how about we return points, if there is order then we should name them major or minor